packaging/generic-rpm/ecs-agent.spec

```diff
@@ -0,0 +1,152 @@
+#
+# Licensed under the Apache License, Version 2.0 (the
+# "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and
+# limitations under the License.
+
+
+%global gobuild_tag generic_rpm
+%global _cachedir %{_localstatedir}/cache
+%global bundled_agent_version %{version}
+%global no_exec_perm 644
+
+%global ecscni_goproject github.com/aws
+%global ecscni_gorepo amazon-ecs-cni-plugins
+%global ecscni_goimport %{ecscni_goproject}/%{ecscni_gorepo}
+%global ecscni_gitrev 55b2ae77ee0bf22321b14f2d4ebbcc04f77322e1
+
+%global vpccni_goproject github.com/aws
+%global vpccni_gorepo amazon-vpc-cni-plugins
+%global vpccni_goimport %{vpccni_goproject}/%{vpccni_gorepo}
+%global vpccni_gitrev a21d3a41f922e14c19387713df66be3e4ee1e1f6
+%global vpccni_gover 1.2
+
+Name:           amazon-ecs-agent
+Version:        1.53
+Release:        1
+License:        Apache 2.0
+Summary:        Amazon Elastic Container Service initialization application
+ExclusiveArch:  x86_64 aarch64
+
+Source0:        sources.tgz
+Source1:        ecs.service
+Source2:        https://%{ecscni_goimport}/archive/%{ecscni_gitrev}/%{ecscni_gorepo}.tar.gz
+Source3:        https://%{vpccni_goimport}/archive/%{vpccni_gitrev}/%{vpccni_gorepo}.tar.gz
+Source4:        amazon-ecs-volume-plugin.service
+Source5:        amazon-ecs-volume-plugin.socket
+
+BuildRequires:  systemd
+Requires:       systemd
+Requires:       iptables
+Requires:       procps
+
+%description
+containerless agent.
+
+%prep
+%setup -c -n amazon-ecs-agent
+
+
+#unpacking the cni plugins
+%setup -T -D -a 2 -q -c -n amazon-ecs-agent/src/github.com/aws
+%setup -T -D -a 3 -q -c -n amazon-ecs-agent/src/github.com/aws
+
+mv %{_builddir}/amazon-ecs-agent/src/github.com/aws/amazon-ecs-cni-plugins* %{_builddir}/amazon-ecs-agent/src/github.com/aws/amazon-ecs-cni-plugins
+mv %{_builddir}/amazon-ecs-agent/src/github.com/aws/amazon-vpc-cni-plugins* %{_builddir}/amazon-ecs-agent/src/github.com/aws/amazon-vpc-cni-plugins
+
+
+%build
+%{_builddir}/amazon-ecs-agent/scripts/build false %{gobuild_tag}
+%{_builddir}/amazon-ecs-agent/scripts/rpm-volumeBuild.sh 
+
+
+# Build the ECS CNI plugins
+export GOPATH=%{_builddir}/amazon-ecs-agent
+
+cd %{_builddir}/amazon-ecs-agent/src/github.com/aws/amazon-ecs-cni-plugins
+LD_ECS_CNI_VERSION="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.Version=$(cat VERSION)"
+ECS_CNI_HASH="%{ecscni_gitrev}"
+LD_ECS_CNI_SHORT_HASH="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.GitShortHash=${ECS_CNI_HASH::8}"
+LD_ECS_CNI_PORCELAIN="-X github.com/aws/amazon-ecs-cni-plugins/pkg/version.GitPorcelain=0"
+go build -a \
```

Comment:
If I understand correctly, the current CNI plugins `cd` into the repo of the cni plugins and run `make` commands that are definied by the maintainers of those repos, rather than us deciding on and running our own go build commands. [here](https://github.com/aws/amazon-ecs-agent/blob/master/scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins#L30) and [here](https://github.com/aws/amazon-ecs-agent/blob/master/scripts/dockerfiles/Dockerfile.buildECSCNIPlugins#L24). Is there some particular reason we are defining them ourselves here? I'm particularly also curious about `-buildmode=pie` being added, since I don't see that in the `make plugins` target of amazon-ecs-cni-plugins: https://github.com/aws/amazon-ecs-cni-plugins/blob/master/Makefile

Reply:
> Is there some particular reason we are defining them ourselves here?

same question. and i'm wondering do we need to build the plugin here at all? it seems to me that we could just let the rpm make target depends on the existing cni-plugins make target, and just use the plugins built from the latter to build the rpm

Reply:
Since the submodules were no longer being used, it was easier to just build and create everything inside of the .spec file. This way, the makefile of the plugin repo won't have to be used and the whole build process for the CNI plugins will be in one spot.

Reply:
is there any advantage to the proposed approach? Is there no way to use submodules as is?